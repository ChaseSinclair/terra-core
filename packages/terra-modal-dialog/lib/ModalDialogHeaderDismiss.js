'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _IconClose = require('terra-icon/lib/icon/IconClose');

var _IconClose2 = _interopRequireDefault(_IconClose);

var _terraButton = require('terra-button');

var _terraButton2 = _interopRequireDefault(_terraButton);

require('./ModalDialogDismiss.scss');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

var propTypes = {
  children: _propTypes2.default.node,
  onClick: _propTypes2.default.func
};

var defaultProps = {
  children: null
};

var ModalDialogDismiss = function ModalDialogDismiss(_ref) {
  var onClick = _ref.onClick,
      customProps = _objectWithoutProperties(_ref, ['onClick']);

  return _react2.default.createElement(
    _terraButton2.default,
    _extends({ variant: 'link', className: 'terra-ModalDialog-header-dismiss', onClick: onClick }, customProps),
    _react2.default.createElement(_IconClose2.default, null)
  );
};

ModalDialogDismiss.propTypes = propTypes;
ModalDialogDismiss.defaultProps = defaultProps;

exports.default = ModalDialogDismiss;