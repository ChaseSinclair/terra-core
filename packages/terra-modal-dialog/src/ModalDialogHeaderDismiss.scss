@import '~terra-mixins';

.terra-ModalDialog-header-dismiss {
  @include terra-padding-horizontal(0, 0);
  border: 0;
  color: $terra-dark;
  cursor: pointer;
  height: 2em;
  margin-bottom: 0.3571rem;
  margin-left: 0.7143rem;
  margin-right: 0.7143rem;
  margin-top: 0.3571rem;
  width: 2em;
  display: table-cell;
}
