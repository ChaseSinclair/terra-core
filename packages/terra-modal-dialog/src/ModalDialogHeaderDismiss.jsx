import React from 'react';
import PropTypes from 'prop-types';
import IconClose from 'terra-icon/lib/icon/IconClose';
import Button from 'terra-button';
import './ModalDialogHeaderDismiss.scss';

const propTypes = {
  children: PropTypes.node,
  onClick: PropTypes.func,
};

const defaultProps = {
  children: null,
};

const ModalDialogDismiss = ({ onClick, ...customProps }) => (
  <Button variant="link" className="terra-ModalDialog-header-dismiss" onClick={onClick} {...customProps} >
    <IconClose />
  </Button>);

ModalDialogDismiss.propTypes = propTypes;
ModalDialogDismiss.defaultProps = defaultProps;

export default ModalDialogDismiss;
