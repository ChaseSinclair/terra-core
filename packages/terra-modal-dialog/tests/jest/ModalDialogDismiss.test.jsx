import React from 'react';
import ModalDialogHeaderDismiss from '../../src/ModalDialogHeaderDismiss';

const modalDialogHeaderDismiss = <ModalDialogHeaderDismiss>This is content</ModalDialogHeaderDismiss>;

it('should shallow an open modal', () => {
  const modal = shallow(modalDialogHeaderDismiss);
  expect(modal).toMatchSnapshot();
});

it('should mount an open modal', () => {
  const modal = mount(modalDialogHeaderDismiss);
  expect(modal).toMatchSnapshot();
});

it('should render an open modal', () => {
  const modal = render(modalDialogHeaderDismiss);
  expect(modal).toMatchSnapshot();
});

it('should have class terra-ModalDialog-header-dismiss', () => {
  const modal = shallow(modalDialogHeaderDismiss);
  expect(modal.prop('className')).toEqual('terra-ModalDialog-header-dismiss');
});
