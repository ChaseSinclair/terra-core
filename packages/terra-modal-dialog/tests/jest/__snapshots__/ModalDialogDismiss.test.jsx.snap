exports[`test should mount an open modal 1`] = `
<ModalDialogDismiss>
  <Button
    className="terra-ModalDialog-header-dismiss"
    isBlock={false}
    isCompact={false}
    isDisabled={false}
    isReversed={false}
    variant="link">
    <button
      aria-disabled={false}
      className="terra-Button terra-Button--link terra-ModalDialog-header-dismiss"
      disabled={false}>
      <IconClose
        data-name="Layer 1"
        viewBox="0 0 48 48"
        xmlns="http://www.w3.org/2000/svg">
        <IconBase
          ariaLabel={null}
          data-name="Layer 1"
          focusable={false}
          height="1em"
          isBidi={false}
          isSpin={false}
          viewBox="0 0 48 48"
          width="1em"
          xmlns="http://www.w3.org/2000/svg">
          <svg
            aria-hidden="true"
            className="terra-Icon"
            data-name="Layer 1"
            focusable={false}
            height="1em"
            viewBox="0 0 48 48"
            width="1em"
            xmlns="http://www.w3.org/2000/svg">
            <path
              d="M28.2 24L42.9 9.1 40.8 7l-1.7-1.6-.4-.5L24 19.7 9.4 4.9 7.2 7 5.6 8.6l-.5.5L19.8 24 5.1 38.9 7.2 41l1.7 1.6.5.5L24 28.3l14.7 14.8.4-.5 1.7-1.6 2.1-2.1L28.2 24z" />
          </svg>
        </IconBase>
      </IconClose>
    </button>
  </Button>
</ModalDialogDismiss>
`;

exports[`test should render an open modal 1`] = `
<button
  aria-disabled="false"
  class="terra-Button terra-Button--link terra-ModalDialog-header-dismiss">
  <svg
    aria-hidden="true"
    class="terra-Icon"
    data-name="Layer 1"
    focusable="false"
    height="1em"
    viewbox="0 0 48 48"
    width="1em"
    xmlns="http://www.w3.org/2000/svg">
    <path
      d="M28.2 24L42.9 9.1 40.8 7l-1.7-1.6-.4-.5L24 19.7 9.4 4.9 7.2 7 5.6 8.6l-.5.5L19.8 24 5.1 38.9 7.2 41l1.7 1.6.5.5L24 28.3l14.7 14.8.4-.5 1.7-1.6 2.1-2.1L28.2 24z" />
  </svg>
</button>
`;

exports[`test should shallow an open modal 1`] = `
<Button
  className="terra-ModalDialog-header-dismiss"
  isBlock={false}
  isCompact={false}
  isDisabled={false}
  isReversed={false}
  variant="link">
  <IconClose
    data-name="Layer 1"
    viewBox="0 0 48 48"
    xmlns="http://www.w3.org/2000/svg" />
</Button>
`;
